modules/scan/class-admin-sidebar-link.php

```diff
@@ -0,0 +1,192 @@
+<?php
+/**
+ * A class that adds a scan and backup link to the admin sidebar.
+ *
+ * @package automattic/jetpack
+ */
+
+namespace Automattic\Jetpack\Scan;
+
+use Automattic\Jetpack\Redirect;
+use Jetpack_Core_Json_Api_Endpoints;
+
+/**
+ * Class Main
+ *
+ * Responsible for showing the link if available.
+ *
+ * @package Automattic\Jetpack\Scan
+ */
+class Admin_Sidebar_Link {
+
+	const SCHEDULE_ACTION_HOOK = 'jetpack_scan_refresh_states_event';
+
+	/**
+	 * The singleton instance of this class.
+	 *
+	 * @var Admin_Sidebar_Link
+	 */
+	protected static $instance;
+
+	/**
+	 * Used to check if we need to schedule the refresh or we need to do it.
+	 *
+	 * @var boolean | null
+	 */
+	private $schedule_refresh_checked;
+
+	/**
+	 * Get the singleton instance of the class.
+	 *
+	 * @return Admin_Sidebar_Link
+	 */
+	public static function instance() {
+		if ( ! isset( self::$instance ) ) {
+			self::$instance = new Admin_Sidebar_Link();
+			self::$instance->init_hooks();
+		}
+
+		return self::$instance;
+	}
```

Comment:
@enejb, I'm curious why we use the singleton pattern here?

Reply:
So that other plugin authors can unhook the hooks that we added. Otherwise there is no way of doing this without adding global variables.